### ISSUE TITLE
Cairo: Structure code according to documentation

### ISSUE THREAD

**Issue by badcel** [gircore/gir.core#590]
As an example we have currently those functions in [context.cs](https://github.com/gircore/gir.core/blob/38129d6a0b2405b32225756741cfc3e5e0eb55eb/src/Libs/cairo-1.0/Internal/Records/Context.cs#L5):

```CS
[DllImport(DllImportOverride.CairoLib, EntryPoint = "cairo_rectangle")]
public static extern void Rectangle(ContextHandle cr, double x, double y, double width, double height);

[DllImport(DllImportOverride.CairoLib, EntryPoint = "cairo_set_font_size")]
public static extern void SetFontSize(ContextHandle cr, double size);
```

According to the documentation [cairo_rectangle](https://cairographics.org/manual/cairo-Paths.html#cairo-rectangle) is inside some "Path" area.

On the other hand the documentation for [cairo_set_font_size](https://cairographics.org/manual/cairo-text.html#cairo-set-font-size) suggests an area of "Text".

What about restructuring the code (internal and public) to keep things a little more separated. The `DllImport` calls could move into classes called "Path" and "Text" respectively.

For the public API I would imagine something like:

Path.cs:
```CS
public static class Path
{
    public static void Rectangle(this Context context, double x, double y, double width, double height)
     => Internal.Path.Rectangle(context.Handle, x, y, width, height);
}
```

Text.cs:
```CS
public static class Text
{
    public static void SetFontSize(this Context context, double size)
     => Internal.Text.SetFontSize(context.Handle, size);
}
```

Using the public API would not be different for the user as the methods of `Text` and `Path` are just extension methods which extend the context. But from a coding point of view they are separated into logical "clusters". Additionally this allows a user to write the code "the othery way around", too. See the following (bad) example:

```CS
private void SetMyFontSizeTo10(Context context)
{
    //Both calls in the same code being executed:
    Text.SetFontSize(context, 10);
    context.SetFontSize(10);
}
```

@cameronwhite what do you think?

**Comment by cameronwhite** [comment]
Splitting the Cairo.Context methods into logical groups definitely sounds okay to me on the internal side of things, since there will be a ton of methods 👍 
I'm not 100% sure about doing this publicly - e.g. for the `Path` example, there is actually a `Cairo.Path` object (`cairo_path_t`) so that may cause some confusion

**Comment by badcel** [comment]
Good catch! Actually the name is not `Path` but `Paths`. I wonder if this was somehow intended to avoid naming conflicts by the cairo devs. So this could still work.

It looks like `Transformations` would be the 3rd candidate as a container for some functions.

The alternative would be to put the code in the `Context` class but create several `partial` files e.g. `Context.Paths.cs`, `Context.Text.cs`. `Context.Transformations.cs`. Perhaps this is even better to avoid confusing developers with multiple ways of doing the same thing.

**Comment by cameronwhite** [comment]
Yeah I'm leaning towards the latter suggestion of just splitting the implementation across files for the different logical groups, but keeping the API just on `Cairo.Context`
